module/VuFindApi/src/VuFindApi/Controller/ApiTrait.php

```
@@ -202,4 +224,39 @@ protected function output($data, $status, $httpCode = null, $message = '')
         $headers->addHeader($contentTypeHeader);
         return $response;
     }
+
+    /**
+     * Init API key settings
+     *
+     * @param array $settings API key settings from config.ini
+     *
+     * @return void;
+     */
+    protected function initApiKeySettings(array $settings): void
+    {
+        $this->developerSettingsService = $this->getService(DeveloperSettingsService::class);
+        if ($field = $settings['header_field'] ?? null) {
+            $this->apiKeyHeaderField = $field;
+        }
+    }
+
+    /**
+     * Check request for API key if mode is not set to disabled.
+     *
+     * @return bool
+     */
+    protected function checkRequestForApiKey(): bool
+    {
+        return $this->developerSettingsService->isApiKeyAllowed($this->getHeader($this->apiKeyHeaderField));
```

Should we throw an exception if this gets called before $this->developerSettingsService has been populated?